### ISSUE TITLE
ScrapeConfig: unknown Kubernetes SD role \"Pod\"" when using kubernetesSDConfigs

### ISSUE THREAD

**Issue by rgomezceis** [prometheus-operator/prometheus-operator#6885]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When deploying scrape config with kubernetesSDConfigs, role is not parsing correctly to prometheus yaml format. Maybe is looking for lowercase but ScrapeConfig kubernetesSDConfigs definition does not support lowercase letters (Pod)

### Prometheus Operator Version

```shell
0.76.0
```

### Kubernetes Version

```yaml
v1.30.2
```

### Kubernetes Cluster Type

Other (please comment)

### How did you deploy Prometheus-Operator?

prometheus-operator/kube-prometheus

### Manifests

```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: ScrapeConfig
metadata:
  name: veeam-k10-pods
  namespace: monitoring
  labels:
    prometheus: system-monitoring-prometheus
    app.kubernetes.io/name: scrape-config-k10-pods
spec:
  jobName: k10-pods
  kubernetesSDConfigs:
    - role: Pod
      namespaces:
        ownNamespace: true
      selectors:
        - role: Pod
          label: "component=executor"
  relabelings:
    - action: labelmap
      regex: __meta_kubernetes_pod_label_(.+)
    - sourceLabels: [__meta_kubernetes_pod_container_port_number]
      action: keep
      regex: 8\d{3}
```

### prometheus-operator log output

```shell
ts=2024-08-29T08:24:05.845Z caller=main.go:549 level=error msg="Error loading config (--config.file=/etc/prometheus/config_out/prometheus.env.yaml)" file=/etc/prometheus/config_out/prometheus.env.yaml err="parsing YAML file /etc/prometheus/config_out/prometheus.env.yaml: unknown Kubernetes SD role \"Pod\""
```

**Comment by simonpasquier** [comment]
Thanks for the report, we've introduced the regression in 38900ced62. The [fix](https://github.com/prometheus-operator/prometheus-operator/blob/5da3c988dd0ce038a55271b2c6ae355b7d971d4f/pkg/prometheus/promcfg.go#L2874) should be *easy* but needs to be done against the release-0.76.
cc @mviswanathsai  @nicolastakashi

**Comment by nicolastakashi** [comment]
@mviswanathsai can you take the fix or should I?

**Comment by simonpasquier** [comment]
Fixed by #6896
